src/rooms/list/v1.rs

```diff
@@ -0,0 +1,191 @@
+//! [GET /_synapse/admin/v1/rooms](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/rooms.md#list-room-api)
+use js_int::UInt;
+use ruma::{
+    api::ruma_api,
+    events::room::{
+        guest_access::GuestAccess, history_visibility::HistoryVisibility, join_rules::JoinRule,
+    },
+    serde::StringEnum,
+    RoomAliasId, RoomId, UserId,
+};
+use serde::{Deserialize, Serialize};
+
+/// Enum to define the sorting method of rooms.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, StringEnum)]
+#[ruma_enum(rename_all = "snake_case")]
+pub enum RoomSortOrder {
+    /// Sort by name alphabetical
+    Name,
+
+    /// Sort by canonical alias
+    CanonicalAlias,
+
+    /// Sort by joined members
+    JoinedMembers,
+
+    /// Sort by joined local members
+    JoinedLocalMembers,
+
+    /// Sort by version
+    Version,
+
+    /// Sort by creator
+    Creator,
+
+    /// Sort by encryption
+    Encryption,
+
+    /// Sort by feaeratable
+    Federatable,
+
+    /// Sort by public
+    Public,
+
+    /// Sort by join rules
+    JoinRules,
+
+    /// Sort by guest access
+    GuestAccess,
+
+    /// Sort by history visibility
+    HistoryVisibility,
+
+    /// Sort by state events
+    StateEvents,
+    #[doc(hidden)]
+    _Custom(String),
+}
+
+/// Enum to define the sort order direction.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, StringEnum)]
+pub enum SortDirection {
```

Comment:
We actually have this enum in ruma-client-api, but with a slightly different definition. Maybe we shouldn't deduplicate until it's moved to ruma-common though.